vinculacion/_linking_algorithm.py

```
@@ -0,0 +1,575 @@
+            if hash_val not in hash_to_uofs:
+                hash_to_uofs[hash_val] = []
+            hash_to_uofs[hash_val].append(uof)
+            
+        # 4. Find matches based on up hashes
+        print("\n🔍 Finding hash-based matches...")
+        exact_matches, ambiguous_matches = self._find_hash_matches(up_hashes, hash_to_uofs)
+
+        exact_matches_df = pd.DataFrame(exact_matches)
+        ambiguous_matches_df = pd.DataFrame(ambiguous_matches)
+        
+        print(f"\n📊 MATCHING ROUND RESULTS:")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)